src/emqx_shared_sub.erl

```diff
@@ -264,23 +267,31 @@ pick(sticky, ClientId, SourceTopic, Group, Topic, FailedSubs) ->
             {fresh, Sub0};
         false ->
             %% randomly pick one for the first message
-            {Type, Sub} = do_pick(random, ClientId, SourceTopic, Group, Topic, [Sub0 | FailedSubs]),
-            %% stick to whatever pick result
-            erlang:put({shared_sub_sticky, Group, Topic}, Sub),
-            {Type, Sub}
+            FailedSubs1 = maps_put_new(FailedSubs, Sub0, inactive),
+						case do_pick(random, ClientId, SourceTopic, Group, Topic, FailedSubs1) of
```

Comment:
nit: use space instead tab for indent, otherwise it may be strange in others editor